packages/forms/src/styled/input-group/StyledAppend.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled, { ThemeProps, DefaultTheme, css } from 'styled-components';
+import { retrieveComponentStyles, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+
+const COMPONENT_ID = 'forms.input_group.append';
+
+const borderStyles = (props: ThemeProps<DefaultTheme>) => {
+  const startDirection = props.theme.rtl ? 'right' : 'left';
+  const endDirection = props.theme.rtl ? 'left' : 'right';
+
+  return css`
+    /* stylelint-disable */
+    margin-${startDirection}: -${props.theme.borderWidths.sm};
+
+    & > * {
+      border-top-${startDirection}-radius: 0;
+      border-bottom-${startDirection}-radius: 0;
+    }
+
+    &:not(:last-child) > * {
+      border-top-${endDirection}-radius: 0;
+      border-bottom-${endDirection}-radius: 0;
+    }
+    /* stylelint-enable */
+  `;
+};
+
+export const StyledAppend = styled.div.attrs({
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+})`
+  display: flex;
+  z-index: 1;
+
+  &:hover,
+  &:active {
+    z-index: 2;
+  }
+
+  &:disabled {
+    z-index: -1;
+  }
```

Comment:
Disabled styling doesn't appear to be playing as expected. This div cannot be disabled. But when it's child button is disabled, the hover styling gets wonky. Styling should reflect, to some degree, the disabled styling seen with https://zendeskgarden.github.io/react-components/buttons/#!/Basic/9.

Reply:
Could be that the example should be utilizing a `ButtonGroup` within the `InputGroup` in order to get the styling right?

Reply:
> disabled styling doesn't appear to be playing as expected

I was accidentally applying `z-index` to the wrapping div. This now targets the child elements correctly.

Reply:
I'm not sure using a `ButtonGroup` would work here. It includes semantic roles and keyboard interaction that diverges from this pattern.